src/Interfaces/Typeahead/Interface.purs

```diff
@@ -0,0 +1,236 @@
+  , keepOpen: r.keepOpen
+  , itemToObject: \a -> a
+  , asyncConfig: Nothing
+  , render: renderMulti
+      [ HP.placeholder r.placeholder ]
+      (renderFuzzy <<< match false identity "")
+      (defRenderContainer renderFuzzy)
+  }
+  where
+    renderFuzzy = span_ <<< boldMatches r.key
+
+
+-- A helper function for parsing strings to status types
+getStatus
+  :: String
+  -> String
+  -> Array (Object String)
+  -> RemoteData String (Array (Object String))
+getStatus s e i
+  | s == "success" || s == "Success" = Success i
+  | s == "failure" || s == "Failure" = Failure e
+  | s == "loading" || s == "Loading" = Loading
+  | s == "notAsked" || s == "NotAsked" = NotAsked
+  | otherwise = Success i
+
+mountMultiTypeahead :: EffectFn2 HTMLElement ExternalInput (Interface MessageVariant QueryRow)
+mountMultiTypeahead = mkEffectFn2 \el ext -> do
+  ioVar <- AVar.empty
+  launchAff_ do
+    io <- runUI multi (externalInputToMultiInput ext) el
+    AffAVar.put io ioVar
+  pure
+    { subscribe: mkSubscription ioVar convertMultiToMessageVariant
+    , remove: mkEffectFn1 \obj -> Promise.fromAff do
+        io <- AffAVar.read ioVar
+        io.query $ Remove obj unit
+    , removeAll: Promise.fromAff do
+        io <- AffAVar.read ioVar
+        io.query $ RemoveAll unit
+    , triggerFocus: Promise.fromAff do
+        io <- AffAVar.read ioVar
+        io.query $ TriggerFocus unit
+    , search: mkEffectFn1 \str -> Promise.fromAff do
+        io <- AffAVar.read ioVar
+        io.query $ Search str unit
+    , getSelected: Promise.fromAff do
+        io <- AffAVar.read ioVar
+        io.query $ GetSelected identity
+    , replaceSelected: mkEffectFn1 \arr -> Promise.fromAff do
+        io <- AffAVar.read ioVar
+        io.query $ ReplaceSelected arr unit
+    , replaceItems: mkEffectFn3 \status err arr -> Promise.fromAff do
+        io <- AffAVar.read ioVar
+        io.query $ ReplaceItems (getStatus status err arr) unit
+    , reset: Promise.fromAff do
+        io <- AffAVar.read ioVar
+        io.query $ Reset unit
+    }
+
+
+mountSingleTypeahead :: EffectFn2 HTMLElement ExternalInput (Interface MessageVariant QueryRow)
+mountSingleTypeahead = mkEffectFn2 \el ext -> do
+  ioVar <- AVar.empty
+  launchAff_ do
+    io <- runUI single (externalInputToSingleInput ext) el
+    AffAVar.put io ioVar
+  pure
+    { subscribe: mkSubscription ioVar convertSingleToMessageVariant
+    , remove: mkEffectFn1 \obj -> Promise.fromAff do
+        io <- AffAVar.read ioVar
+        io.query $ Remove obj unit
+    , removeAll: Promise.fromAff do
+        io <- AffAVar.read ioVar
+        io.query $ RemoveAll unit
+    , triggerFocus: Promise.fromAff do
+        io <- AffAVar.read ioVar
+        io.query $ TriggerFocus unit
+    , search: mkEffectFn1 \str -> Promise.fromAff do
+        io <- AffAVar.read ioVar
+        io.query $ Search str unit
+    , getSelected: Promise.fromAff do
+        io <- AffAVar.read ioVar
+        res <- io.query $ GetSelected identity
```

Comment:
I'm not totally clear on why `identity` works as the `f item -> a` function here.

Reply:
Check out the implementation of `eval (GetSelected f)` in the typeahead -- I think that'll make it more clear.

Reply:
```purescript
      GetSelected reply -> do
        { selected } <- getState
        pure $ reply selected
```